src/viur/core/modules/user.py

```diff
@@ -439,7 +439,7 @@ def pwrecover(self, recovery_key: str | None = None, skey: str | None = None, *a
             )
 
         # If the account is locked or not yet validated, abort the process.
-        if not self._user_module.is_active(user_skel):
+        if self._user_module.is_active(user_skel) is False:
```

Comment:
In this case I would keep the old behaviour. It's necessary to include the status in the viewSkel, otherwise you can't determine if it's allowed and it should fail.